src/Generator/ParameterGenerator.php

```diff
@@ -163,14 +163,20 @@ public function __construct(
     }
 
     /**
-     * @param  string $type
+     * @param  mixed $type
```

Comment:
I wouldn't allow `array`. Pass in a `TypeGenerator` if needed: arrays are squishy. Let's avoid stringly-typed and array-typed code ;-)